src/core/tile.rs

```diff
@@ -23,19 +23,19 @@ use serde::{Deserialize, Serialize};
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[derive(Clone, PartialEq, Debug)]
 pub struct Tile<T: Clone> {
-    pub data: Option<Box<T>>,
+    pub data: T,
 }
 
 impl<T: Default + Clone> Tile<T> {
     pub fn new(data: Option<T>) -> Tile<T> {
         Tile {
-            data: data.map(|value| Box::new(value)),
+            data: data.unwrap_or_default(),
```

Comment:
ah, yes, this looks much more elegant

Reply:
That's just a side effect of defining the API bounds. The array and HashMaps will store them on the heap so no need to put them in a box. and if someone wants to, they can on their struct. That map function is actually the way you can unwrap an option and construct it into a box according to Clippy.